onnxruntime/core/session/inference_session.h

```diff
@@ -132,8 +132,12 @@ class InferenceSession {
   explicit InferenceSession(const SessionOptions& session_options,
                             const Environment& session_env);
 
-#if !defined(ORT_MINIMAL_BUILD)
+  explicit InferenceSession(const SessionOptions& session_options,
+                            const Environment& session_env,
+                            onnxruntime::concurrency::ThreadPool* intra_op_thread_pool,
```

Comment:
OrtEnv already has a global thread pool. Is this change really needed?

Reply:
The global thread pool cannot be used in winml, as there is no global environment object exposed. We have a device object with is the natural place to own these settings.